File: Examples/VisualBasic/Loading-and-Saving/SaveToStream.vb
Imports System.IO
Imports Aspose.Note

Namespace Loading_Saving
    Public Class SaveToStream
        Public Shared Sub Run()
            ' ExStart:SaveToStream
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_LoadingAndSaving()

            ' Load the document into Aspose.Note.
            Dim doc As New Document(dataDir & Convert.ToString("Aspose.one"))

            Dim dstStream As New MemoryStream()
            doc.Save(dstStream, SaveFormat.Pdf)

            ' Rewind the stream position back to zero so it is ready for next reader.
            dstStream.Position = 0
            ' ExEnd:SaveToStream
            Console.WriteLine(vbLf & "OneNote document saved successfully to stream.")
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Loading-and-Saving/SaveToStream.cs
using System.IO;
using Aspose.Note;
using System;

namespace Aspose.Note.Examples.CSharp.Loading_Saving
{
    public class SaveToStream
    {
        public static void Run()
        {
            // ExStart:SaveToStream
            // ExFor:Document
            // ExFor:Document.Save(System.IO.Stream, Aspose.Note.SaveFormat)
            // ExSummary:Shows how to save a document to a stream.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_LoadingAndSaving();

            // Load the document into Aspose.Note.
            Document doc = new Document(dataDir + "Aspose.one");

            MemoryStream dstStream = new MemoryStream();
            doc.Save(dstStream, SaveFormat.Pdf);

            // Rewind the stream position back to zero so it is ready for next reader.
            dstStream.Seek(0, SeekOrigin.Begin);
            
            // ExEnd:SaveToStream

            Console.WriteLine("\nOneNote document saved successfully to stream.");
        }
    }
}